packages/eas-cli/src/user/expoSsoLauncher.ts

```diff
@@ -0,0 +1,80 @@
+import openBrowserAsync from 'better-opn';
+import http from 'http';
+import { Socket } from 'node:net';
+import querystring from 'querystring';
+
+import Log from '../log';
+
+export async function initiateAuthFlowAsync(options: {
+  expoWebsiteUrl: string;
+  serverPort: number;
+}): Promise<string> {
+  const { expoWebsiteUrl, serverPort } = options;
+  if (!expoWebsiteUrl || !serverPort) {
+    throw new Error('Expo website URL and local server port are required.');
+  }
+  const scheme = 'http';
+  const hostname = 'localhost';
+  const path = '/auth/callback';
+  const redirectUri = `${scheme}://${hostname}:${serverPort}${path}`;
+
+  const buildExpoSsoLoginUrl = (): string => {
+    const data = {
+      app_redirect_uri: redirectUri,
+    };
+    const params = querystring.stringify(data);
+    return `${expoWebsiteUrl}/sso-login?${params}`;
+  };
+
+  // Start server and begin auth flow
+  const executeAuthFlow = (): Promise<string> => {
+    return new Promise<string>(async (resolve, reject) => {
+      const connections = new Set<Socket>();
+
+      const server = http.createServer(
+        (request: http.IncomingMessage, response: http.ServerResponse) => {
+          try {
+            if (!(request.method === 'GET' && request.url?.includes('/auth/callback'))) {
+              throw new Error('Unexpected SSO login response.');
+            }
+            const url = new URL(request.url, `http:${request.headers.host}`);
+            const sessionSecret = url.searchParams.get('session_secret');
+
+            if (!sessionSecret) {
+              throw new Error('Request missing session_secret search parameter.');
+            }
+            resolve(sessionSecret);
+            response.writeHead(200, { 'Content-Type': 'text/plain' });
+            response.write(`Thank you, website login has completed. You can now close this tab.`);
```

Comment:
```suggestion
            response.write(`Website login has completed. You can now close this tab.`);
```

(I think we're trying to reduce the amount of "please" and "thank you" in our messaging)

Reply:
Done